python_ta/typecheck/base.py

```diff
@@ -129,42 +133,83 @@ def lookup_method(name, caller_type, *args):
     return TYPE_SIGNATURES[caller_origin][name]
 
 
+def make_set(node):
+    node.parent = node
+    # node.rank = 0
+
+
+def _find_rep(node):
+    while node.parent != node:
+        node = node.parent
+    return node
+
+
+def _union(node1, node2):
+    rep1 = _find_rep(node1)
+    rep2 = _find_rep(node2)
+    rep2.parent = rep1
```

Comment:
Let's make this a bit better. Pick as parent the node that has a *concrete* type, if it exists.